### PR TITLE
feat(apps.plugin): collect context switches 

### PR DESCRIPTION
**Pull request by ilyam8**
##### Summary

Fixes: #14999

- [X] collect voluntary/non-voluntary context switches.
- [x] add them to functions.

##### Test Plan

Check the new metrics:
- On the dashboard (`cpu` family).
- In Functions.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Comment by ilyam8** [comment]
Seems to be working, the only problem I see is the bar chart. For some reason `system` and `kernel` groups are not in there despite both of them being in the table below. Any ideas on what could be the problem, @ktsaou ?
Screenshot 2023-05-03 at 23 13 18

File: collectors/apps.plugin/apps_plugin.c [hunk 1]
```diff
@@ -4610,6 +4678,10 @@ static void apps_plugin_function_processes(const char *transaction, char *functi
         add_table_field(wb, "CSysCPU", "Children System CPU Time (100% = 1 core)", false, "bar-with-integer", "bar", "number", 2, "%", CSysCPU_max, "descending", true, false, false, NULL, "sum", true);
         add_table_field(wb, "CGuestCPU", "Children Guest CPU Time (100% = 1 core)", false, "bar-with-integer", "bar", "number", 2, "%", CGuestCPU_max, "descending", true, false, false, NULL, "sum", true);
 
+        // CPU context switches
+        add_table_field(wb, "vCtxSwitch", "Voluntary Context Switches", false, "bar-with-integer", "bar", "number", 2, "switches/s", VoluntaryCtxtSwitches_max, "descending", true, false, false, NULL, "sum", true);
```
Comment (thiagoftsm) [inline collectors/apps.plugin/apps_plugin.c:4682]: Why only `switches/s`? The unit of the chart is `context switch`.

Comment (ilyam8) [reply collectors/apps.plugin/apps_plugin.c:4686]: The unit of both charts is `switches/s`. Where did you find `context switch`?

**Review by thiagoftsm — CHANGES_REQUESTED** [review]
@ilyam8 ,

While I was testing this PR, I observed that our `system.ctxt`:

```sh
"system.ctxt": 		{
			"id": "system.ctxt",
			"name": "system.ctxt",
			"type": "system",
			"family": "processes",
			"context": "system.ctxt",
			"title": "CPU Context Switches (system.ctxt)",
			"priority": 800,
			"plugin": "proc.plugin",
			"module": "/proc/stat",
			"units": "context switches/s",
			"data_url": "/api/v1/data?chart=system.ctxt",
			"chart_type": "line",
			"duration": 4006,
			"first_entry": 1683203178,
			"last_entry": 1683207183,
			"update_every": 1,
			"dimensions": {
				"switches": { "name": "switches" }
			},
			"chart_variables": {
			},
			"green": null,
			"red": null,
			"alarms": {

			},
			"chart_labels": {
				"_collect_plugin":"proc.plugin",
				"_collect_module":"/proc/stat"
			},
			"functions": {
			}
		},
```

Has as unit `context switches`, so it is no a minor to modify the `units` here

**Comment by thiagoftsm** [comment]
Except for the units, plugin is working as expected on `Linux` and `FreeBSD`.

File: collectors/apps.plugin/apps_plugin.c [hunk 2]
```diff
@@ -3822,6 +3873,22 @@ static void send_charts_updates_to_netdata(struct target *root, const char *type
         APPS_PLUGIN_FUNCTIONS();
     }
 
+#ifndef __FreeBSD__
```
[the remark this answers is not in the document]
Comment (thiagoftsm) [reply collectors/apps.plugin/apps_plugin.c:3876]: MAC OS, is `__FreeBSD__` enough for both?

**Comment by thiagoftsm** [comment]
> > Has as unit context switches, so it is not a minor to modify the units here
> 
> I don't think that I need to modify units because that chart has `context switches/s`.

Like here. :) We have context switch per application.

**Comment by ilyam8** [comment]
@thiagoftsm I have no plans to change the unit to `context switches/s`:
 - it is too long and gets shortened on the Cloud UI (unreadable).
 - I think it is ok to have it as  switches/s because in the context of the chart/metric name, it is pretty clear what that is.

File: collectors/apps.plugin/apps_plugin.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilyam8) [reply collectors/apps.plugin/apps_plugin.c:3876]: No, it is not enough. But apps.plugin has **never** been working on macOS (read as there is no apps.plugin on macOS).

**Review by thiagoftsm — APPROVED** [review]

**Pull request merged**